task-launcher/src/tasks/shared/helpers/taskStore.js

```diff
@@ -0,0 +1,99 @@
+ * @property {number} trialNumTotal - Counter for total trials in the experiment, starting at 0.
+ * @property {boolean} currentTrialCorrect - Indicates if the current trial is correct, default is true.
+ * @property {number} numIncorrect - Counter for incorrect responses, starting at 0.
+ * @property {number} totalCorrect - Counter for total correct trials, starting at 0.
+ * @property {Array} correctItems - List of correct items, starting as an empty array.
+ * @property {string} audioFeedback - Audio feedback to use, default is 'neutral'.
+ * @property {boolean} skipInstructions - Whether to skip instructions, default is true.
+ * @property {string} corpusId - Name of the corpus file.
+ * @property {string} buttonLayout - Layout of the buttons, default is 'default'.
+ * @property {string} task - Name of the task, default is 'egma-math'.
+ * @property {number} maxIncorrect - Maximum number of incorrect trials, default is 3.
+ * @property {boolean} keyHelpers - Whether to use keyboard helpers, default is true.
+ * @property {boolean} storeItemId - Whether to store the item ID, default is false.
+ * @property {boolean} isRoarApp - Whether the app is running in ROAR mode, default is false.
+ * @property {boolean} maxTimeReached - Whether the max time has been reached, default is false.
+ * ------- Added after config is parsed -------
+ * @property {number} totalTrials - Counter for total trials in the experiment, starting at 0.
+ * @property {Object} corpora - Object containing the corpus data (stimulus).
+ * @property {Object} translations - Object containing the translations.
+ * @property {Object} nextStimulus - Object containing the next stimulus.
+ * ------- AFC and SDS only -------
+ * @property {string} target - Target item.
+ * @property {Array} choices - List of choices.
+ * ------- AFC only -------
+ * @property {boolean} skipCurrentTrial - Whether to skip the current trial, default is false.
+ * @property {number} correctResponseIdx - Index of the correct response, starting at 0.
+ * ------- Math only -------
+ * @property {Array} nonFractionSelections - List of non-fraction selections.
+ */
+
+/**
+ * Store for managing task state. For all tasks.
+ * 
+ * @type {import('store2').StoreAPI & (() => TaskStore)}
+ */
+export const taskStore = store.page.namespace('taskStore');
+
+export const setTaskStore = (config) => {
+  taskStore({
+    itemSelect: 'mfi',
+    trialNumSubtask: 0,
+    trialNumTotal: 0,
+    currentTrialCorrect: false,
+    numIncorrect: 0,
+    // For ROAR syntax (TROG)
+    totalCorrect: 0,
+    correctItems: [],
+    // -----
+    audioFeedback: config.audioFeedback,
+    skipInstructions: config.skipInstructions,
+    corpusId: config.corpusId,
+    buttonLayout: config.buttonLayout,
+    task: config.task,
+    maxIncorrect: config.maxIncorrect,
+    keyHelpers: config.keyHelpers,
+    storeItemId: config.storeItemId,
+    isRoarApp: config.isRoarApp,
+    maxTimeReached: false,
+  });
+};
```

Comment:
This is great. Now we have a centralized location for global config. In the future, when ts is introduced we can restricting writing random things to this store and pollute it unnecessarily